docs/troubleshooting.rst

```diff
@@ -0,0 +1,122 @@
+.. _Troubleshooting:
+
+Troubleshooting
+===============
+
+Kuma has many components. Even core developers need reminders of how to keep
+them all working together. This doc outlines some problems and potential
+solutions running Kuma.
+
+
+Kuma "Reset" script
+-------------------
+
+If you're using the vagrant vm, we have a script that attempts to fix some of
+the most common problems with Kuma that are described below.
+
+`kuma-vagrant-reset.sh <https://gist.github.com/openjck/b69445fa3e34e1780377>`_
+
+
+.. _Running individual processes:
+
+Running individual processes
+----------------------------
+
+It is usually easier to see and debug problems if you run MDN processes
+individually, instead of running them via ``foreman``. You can run each process
+exactly as it is listed in ``Procfile``
+ 
+-  ``runserver`` - runs the Django development server
+
+-  ``celery worker`` - runs the celery worker process for tasks
+
+-  ``celerycam`` - stores a snapshot of celery tasks to display in admin site
+
+-  ``kumascript`` - runs the node.js process for KumaScript macros
+
+-  ``stylus`` - runs a process to compile all ``.styl`` changes into ``.css``
+
+
+Errors after switching branches
+-------------------------------
+
+-  If you are using a vm, you should occasionally re-run the Puppet setup,
+   especially after updating code with major changes. This will ensure that the
+   VM environment stays up to date with configuration changes and installation
+   of additional services.
+
+   -  On the Host::
+
+          vagrant provision
+
+   -  Inside the VM::
+
+          sudo puppet apply /home/vagrant/src/puppet/manifests/dev-vagrant.pp
+
+-  If you see ``ImportError:`` errors, you may need to update your git
+   submodules and/or clean out your ``*.pyc`` files to make sure python has all
+   the latest files it needs.::
+
+       git submodule update --init
+       find . -name "*.pyc" | xargs rm -f
+
+-  If you see ``DatabaseError: (1146, "Table '...' doesn't exist")`` errors,
+   you probably need to run database migrations.::
+
+       python manage.py migrate
+
+   Note: If you are using a vm, this is done when you re-run the Puppet setup.
```

Comment:
Interesting. `provision` certainly runs `migrate`, but if there are errors with `migrate` they will show up as `kuma_south_migrate` errors during provision. I'll add a note to the vagrant troubleshooting section about it.